### DISCUSSION TITLE
Help

### DISCUSSION THREAD

**Discussion by Fridrih909** [ORelio/Minecraft-Console-Client#2229] [Ideas]
Hello. Due to the fact that I do not understand English well, I cannot understand this client. Help create a bot that will log on to the server, solve a captcha of numbers, navigate the compass and then stand still and constantly break blocks. It is necessary that the PCM and LCM are clamped. How to do this? You can even just direct me. Thank you in advance.

**Comment by milutinke** [comment]
Hello,
For solving captchas you can use the [`Map Chat Bot`](https://mccteam.github.io/guide/chat-bots.html#map), to enable it, open the `MinecraftClient.ini`, find `[Map]` section and set `enabled` to `true`.
When you connect to a server, you can use `/maps` command to list and render maps so you can solve capthca.
If the captcha has a really fast time to solve, you can enable `auto_render_on_update` by setting it to `true`.
To list maps you can do: `/maps list`
To render a map you can do: `/maps render <map id here>`

For using the compass, you need to enable [`inventoryhandling`](https://mccteam.github.io/guide/configuration.html#inventoryhandling), and then you can use the [`/useitem`](https://mccteam.github.io/guide/usage.html#useitem) command to use the compass.
When you use the compass, it will open an inventory for you, to list the items in the inventory you've opened, use `/inventory <inventory id here>` command.
More on how to use the [`/inventory`](https://mccteam.github.io/guide/usage.html#inventory) command with examples: [here](https://mccteam.github.io/guide/usage.html#inventory)

**Reply by Fridrih909** [reply]
Thank you very much for your reply. And do not tell me how to make the bot stand still and break the blocks in front of it? This should happen without delay.

**Reply by milutinke** [reply]
> Thank you very much for your reply. And do not tell me how to make the bot stand still and break the blocks in front of it? This should happen without delay.

That feature has not been implemented yet.

**Reply by Fridrih909** [reply]
OK, thank you

**Reply by Fridrih909** [reply]
> > Спасибо вам большое за ваш ответ. А не подскажете, как заставить бота стоять на месте и ломать блоки перед ним? Это должно произойти без промедления.
> 
> Эта функция еще не реализована.

Listen, can you make it just click? After all, there is an auto-attack, how to make it have a minimum delay?

**Reply by milutinke** [reply]
> > > Спасибо вам большое за ваш ответ. А не подскажете, как заставить бота стоять на месте и ломать блоки перед ним? Это должно произойти без промедления.
> > 
> > 
> > Эта функция еще не реализована.
> 
> Listen, can you make it just click? After all, there is an auto-attack, how to make it have a minimum delay?

We can't make the bot just click. The game protocol has different way of sending commands to the server.
The attack sends one command, while for digging blocks there is a different command, if you need the bot to attack faster, you can change [`cooldownseconds`](https://mccteam.github.io/guide/chat-bots.html#cooldownseconds) to your preference.
By the way, now it is possible to mine blocks which you are looking at, download the latest version of the program after #2227 has been merged with the main branch. You can use `/look` to get the block you are looking and and then `/dig` to dig it.

**Reply by Fridrih909** [reply]
As I understand it, now you can automate all this and the bot will stand AFK and constantly mine blocks. OK, thank you for the information.
